src/mbgl/renderer/image_manager.cpp

```diff
@@ -72,29 +98,84 @@ void ImageManager::getImages(ImageRequestor& requestor, ImageRequestPair&& pair)
             }
         }
     }
-    if (isLoaded() || hasAllDependencies) {
-        notify(requestor, std::move(pair));
-    } else {
+
+    if (hasAllDependencies) {
+        notify(requestor, pair);
+    } else if (!isLoaded()) {
         requestors.emplace(&requestor, std::move(pair));
+    } else {
+        checkMissingAndNotify(requestor, std::move(pair));
     }
 }
 
 void ImageManager::removeRequestor(ImageRequestor& requestor) {
     requestors.erase(&requestor);
+    missingImageRequestors.erase(&requestor);
+}
+
+void ImageManager::imagesAdded() {
+    for (auto it = missingImageRequestors.begin(); it != missingImageRequestors.end();) {
+        if (it->second.callbacksRemaining == 0) {
+            notify(*it->first, it->second.pair);
+            missingImageRequestors.erase(it++);
+        } else {
+            it++;
+        }
+    }
+}
+
+void ImageManager::checkMissingAndNotify(ImageRequestor& requestor, const ImageRequestPair& pair) {
+    unsigned int missing = 0;
+    for (const auto& dependency : pair.first) {
+        auto it = images.find(dependency.first);
+        if (it == images.end()) {
+            missing++;
+        }
+    }
+
```

Comment:
nit: empty line

Reply:
done in 3637d15a8fc2be6318c33c22d9cf8b5ecbf00737